### PR TITLE
PV1: Handle `APIVersions` requests

### PR DESCRIPTION
**Pull request by HamzaRaza786**
Worked on Handle Api Version code. Took help, as much isn't available in the challenge. Have created the body struct to show contents of the response body we are sending. Currently the function of body converter is not being used, as need to see how to convert struct into a Vec.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -43,15 +54,47 @@ fn request_headers(mut stream: &TcpStream) -> Headers {
...
     }
 }
+fn construct_response(request_headers: Headers) -> Body {
+    let error_code;
+
+    if !(0..5).contains(&request_headers.request_api_version) {
+        error_code = ErrorCodes::UnsupportedVersion as i16;
+    } else {
+        error_code = 0;
+    }
```
Comment (shirazJafri) [inline src/main.rs]: How about doing something like this - will look more cleaner.

```rust
let supported = (0..5).contains(&request_headers.request_api_version);

let error_code = if !supported { ErrorCodes::UnsupportedVersion as i16 } else { 0 };
```

**Review by shirazJafri — COMMENTED** [review]
We can start looking at `serde` to handle the conversion from `struct` to `Vec`.

File: src/main.rs (hunk 1 above)
Comment (HamzaRaza786) [reply src/main.rs]: would be a nice addition, will fix this.

**Comment by HamzaRaza786** [comment]
> We can start looking at `serde` to handle the conversion from `struct` to `Vec`.

sure.

**Comment by shirazJafri** [comment]
> > We can start looking at `serde` to handle the conversion from `struct` to `Vec`.
> 
> sure.

Can use `serde` because editing `Cargo.toml` is not allowed for the project.

**Review by shirazJafri — APPROVED** [review]
LGTM

**Pull request merged**